TextFormat.md

```diff
@@ -43,43 +38,377 @@ support more human-readable representations, but never at the cost of accurate r
+| `i32.const` | … | `234`, `0xfff7`
+| `i64.const` | … | `234`, `0xfff7`
+| `f64.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `f32.const` | … | `0.1p2`, `infinity`, `nan:0x789`
+| `get_local` | *name* | `$x + 1`
+| `set_local` | *name* `=` *expr* | `$x = 1`
+| `select` | `select` *expr*, *expr*, *expr* | `select 1, 2, $x < $y`
+| `call` | `call` *name* `(`*expr* `,` … `)` | `call $min(0, 2)`
+| `call_import` | `call_import` *name* `(`*expr* `,` … `)` | `call_import $max(0, 2)`
+| `call_indirect` | `call_indirect` *signature-name* `[` *expr* `] (`*expr* `,` … `)` | `call_indirect $foo [1] (0, 2)`
+
+### Arithmetic operators ([described here](AstSemantics#32-bit-integer-operators))
+
+Arithmetic operators use C/JS-like infix and prefix notation.
+
+Add, sub, mul, div, rem, and, or, xor, shl, and shr operators use
+`+`, `-`, `*`, `/`, `%`, `&`, `|`, `^`, `<<`, and `>>`, respectively.
+
+`eq`, `ne`, `lt`, `le`, `gt`, `ge`, operators use
+`==`, `!=`, `<`, `<=`, `>`, `>=`, respectively.
+
+`eqz` and `neg` operators use prefix unary `!` and `-`, respectively.
+
+Signed and unsigned operators are suffixed with `s` and `u`, respectively,
+so `i32.div_s` looks like `/s`, and so on.
+
+These operators use C/JS-style operator precedence rules, except that in order
+to correct [an old mistake](http://www.lysator.liu.se/c/dmr-on-or.html)),
+it is not valid to rely on precedence alone between a `&`, `|`, or `^` operator
+and another operator of similar precedence. Parentheses are required in such
+contexts in order to ensure that the code reads as intended.
+
+All other arithmetic operators use their actual name in a prefix notation,
+such as `f32.sqrt(…)`, with comma-seperated operands.
+
+## Module-level syntax
+
+Type declarations:
+
+`type $ternary of function (i32,i32,i32) : (i32)`
+
+Imports:
+
+`import "print" as $print_int from "global.PrintingStuff" typeof function (f64) : ()`
+
+Function table:
+
+`table [$func0,$func1,$func2,…]`
+
+Exports:
+
+`export memory as "memory"`
+
+`export $func0 as "ThisIsFunc0"`
+
+Function definitions and local variables:
+
+```
+function $func0($x : i32, $y : i32, $y : i32) : (f64) {
+    var $a : i32, $b : i64, $c : f32
+```
 
 # Debug symbol integration
 
+## Debug symbol section
+
 The binary format inherently strips names from functions, locals, globals, etc,
 reducing each of these to dense indices. Without help, the text format must
 therefore synthesize new names. However, as part of the [tooling](Tooling.md)
 story, a lightweight, optional "debug symbol" global section may be defined
 which associates names with each indexed entity and, when present, these names
 will be used in the text format projected from a binary WebAssembly module.
+
+## Code references
+
+In the binary format, references to locations in the code are done through
+byte offsets. In the text format, these references are represented symbolically
+with a special label syntax, prefixed by an `@` sign. These labels have no
```

Comment:
Is there another term for this other than "label" which seems to overlap with "label" as in the label definitions and the earlier label section, unless that is intentional.